constraints/kernel_rom.air

```diff
@@ -0,0 +1,82 @@
+##########################################################################################
+# KERNEL ROM CHIPLET
+##########################################################################################
+#
+# The Kernel ROM chiplet is responsible for tracking execution of kernel (system) calls.
+# It maintains a record of all kernel procedure digests and ensures proper initialization
+# of kernel procedures while preserving privacy about call frequency.
+#
+# The chiplet contains digest values for all kernel procedures and enforces contiguity
+# constraints to ensure proper execution tracking.
+#
+# STATUS: Core constraints implemented, bus constraints not implemented
+#
+# REFERENCES:
+# - Kernel ROM Chiplet: https://0xmiden.github.io/miden-vm/design/chiplets/kernel_rom.html
+##########################################################################################
+
+mod kernel_rom
+
+use utils::*;
+
+##########################################################################################
+# KERNEL ROM CHIPLET TRANSITION CONSTRAINTS
+##########################################################################################
+
+# Enforces the constraints on the kernel ROM chiplet given its columns.
+#
+# Max constraint degree: 3
+ev ker_rom_chiplet_constraints([s4, sfirst, r[4]]) {
+    enf kernel_rom_selector([sfirst]);
+    enf kernel_rom_digest_contiguity([s4, sfirst, r]);
+}
```

Comment:
Aren't we missing a constraint (or 4) to make sure that `sfirst` is set to 1 *only* for the first row of a set of equal roots? I think that currently it's always possible to set `sfirst = 1` for all rows.

This would also be missing in the docs.

Reply:
Ah yes - makes sense!